aten/src/ATen/core/function_schema.cpp

```diff
@@ -16,6 +17,48 @@ std::vector<Argument> FunctionSchema::getCorrectList(SchemaArgType type) const {
   }
 }
 
+bool FunctionSchema::canAliasTypeSetsAlias(const c10::optional<std::vector<TypePtr>> &lhs, const c10::optional<std::vector<TypePtr>> &rhs) const {
+  if (!lhs || !rhs) {
+    return false;
+  }
+  for (const TypePtr& lhsType : *lhs) {
+    for (const TypePtr& rhsType : *rhs) {
+      if (lhsType == rhsType) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+c10::optional<std::vector<TypePtr>> FunctionSchema::getAliasTypeSetContainedTypes(const c10::optional<std::vector<TypePtr>> &aliasTypeSet) const {
+  if (!aliasTypeSet) {
+    return c10::nullopt;
+  }
+  std::unordered_set<TypePtr> containedTypes;
+  std::stack<TypePtr> typeStack;
+  // Push all 1st level contained types into the stack.
+  for (const TypePtr& type: *aliasTypeSet) {
+    for (const TypePtr& containedType : type->containedTypes()){
+      typeStack.push(containedType);
+    }
+  }
+
+  // process all further level contained types.
+  while (!typeStack.empty()) {
+    TypePtr current = typeStack.top();
+    typeStack.pop();
+    if (!containedTypes.count(current)) {
+      for (const TypePtr& containedType : current->containedTypes()) {
+        typeStack.push(containedType);
+      }
+    }
+    containedTypes.insert(current);
+  }
```

Comment:
Nit: we could change this to first check if containedTypes has the type, and only if it does not, insert and iterate through its child types. This would also allow us to use a Vector instead of a unordered_set and avoid the extra conversion.

However, there is also always the saying: Done is better than perfect.